### DISCUSSION TITLE
CustomAuth

### DISCUSSION THREAD

**Discussion by flow2010** [dotnet/yarp#1155] [Q&A] [answered]
Hi There, 
Can you please let me know how to add custom  auth as part proxy pipeline. As part of auth I need to read cookie and validate jwt token and also set cookie. Where should I add middleware class file call so that it can be applied to specific route.

**Comment by Tratcher** [comment, marked as answer]
Option 1) Add a transform to a specific route:
https://microsoft.github.io/reverse-proxy/articles/transforms.html#from-code
https://github.com/microsoft/reverse-proxy/blob/bfb6406e03a342722f0661f0059f63a8c1eedc3d/testassets/ReverseProxy.Code/Startup.cs#L61-L78
Option 2) Add a middleware that applies to all proxy routes, then check for specific route information per request.
- https://microsoft.github.io/reverse-proxy/articles/middleware.html
